qiskit/primitives/containers/observable.py

```diff
@@ -0,0 +1,164 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""
+Sparse container class for an Estimator observable.
+"""
+from __future__ import annotations
+
+from typing import Union, Mapping as MappingType
+from collections.abc import Mapping
+from collections import defaultdict
+from numbers import Complex
+
+from qiskit.quantum_info import Pauli, SparsePauliOp
+
+
+ObservableKey = "tuple[tuple[int, ...], str]"  # pylint: disable = invalid-name
+ObservableKeyLike = Union[Pauli, str, ObservableKey]
+ObservableLike = Union[
+    SparsePauliOp,
+    ObservableKeyLike,
+    MappingType[ObservableKeyLike, complex],
+]
+"""Types that can be natively used to construct a :const:`Observable`."""
+
+
+class Observable(Mapping):
+    """A sparse container for a observable for an :class:`.Estimator` primitive."""
+
+    __slots__ = ("_data", "_num_qubits")
+
+    def __init__(
+        self,
+        data: Mapping[ObservableKey, complex],
```

Comment:
I'm happy to change to float instead of complex (since any hermitian nqubit matrix can be decompose with real coefficients in Pauli basis). Im not sure different exisiting estimators are consistent with enforcing hermitian or not.

Currently for performance its not the inits job to convert Observable-key like, thats the coerce methods job. If you initialize with not an actual observable key validation should fail.

Reply:
In reference to preceding comment, Iean to Key instead of Term since this object is a mapping, and this type is literally the key type for the mapping. In operator speak my first impression of a term would probably be the key * value (coeff).